addons/binding/org.openhab.binding.camera/src/main/java/org/openhab/binding/camera/handler/CameraHandler.java

```diff
@@ -0,0 +1,195 @@
+import org.apache.http.auth.AuthScope;
+import org.apache.http.auth.UsernamePasswordCredentials;
+import org.apache.http.client.AuthCache;
+import org.apache.http.client.CredentialsProvider;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.protocol.HttpClientContext;
+import org.apache.http.impl.auth.BasicScheme;
+import org.apache.http.impl.client.BasicAuthCache;
+import org.apache.http.impl.client.BasicCredentialsProvider;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.eclipse.smarthome.core.library.types.RawType;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link CameraHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Thomas Hartwig - Initial contribution
+ */
+public class CameraHandler extends BaseThingHandler {
+    private Logger logger = LoggerFactory.getLogger(CameraHandler.class);
+    private final AtomicBoolean refreshInProgress = new AtomicBoolean(false);
+    private final AtomicBoolean initialized = new AtomicBoolean(false);
+    private final ExecutorService serviceCached = Executors.newCachedThreadPool();
+    private String urlSnapshot;
+    private String urlUsername = "";
+    private String urlPassword = "";
+
+    public CameraHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (channelUID.getId().equals(CHANNEL_IMAGE)) {
+            if (command.toString().equals("REFRESH")) {
+                refreshData();
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+        if (logger.isDebugEnabled()) {
+            logger.debug("Initialize thing: {}::{}", getThing().getLabel(), getThing().getUID());
+        }
+        Object paramUrl = getConfig().get("urlSnapshot");
+        urlSnapshot = String.valueOf(paramUrl);
+        Object paramUsername = getConfig().get("urlUsername");
+        if (paramUsername != null) {
+            urlUsername = String.valueOf(paramUsername);
+        }
+        Object paramPassword = getConfig().get("urlPassword");
+        if (paramPassword != null) {
+            urlPassword = String.valueOf(paramPassword);
+        }
+        try {
+            Object param = getConfig().get("poll");
+            CONFIGURATION_POLLTIME_MS = (int) (Double.parseDouble(String.valueOf(param)) * 1000);
```

Comment:
I suppose the type of this is a BigDecimal, you can just cast it and ask the `intValue` of it. Maybe a null check could be wise :-)

Also do not overwrite constants, store the polltime here in this class and just the constant as the default value if needed.

Reply:
Changed according to polltime fix.